pkg/image/importer/importer.go

```diff
@@ -129,6 +132,11 @@ func (i *ImageStreamImporter) importImages(ctx gocontext.Context, retriever Repo
 		repoName := defaultRef.RepositoryName()
 		registryURL := defaultRef.RegistryURL()
 
+		if !importer.isAllowedRegistry(defaultRef.RegistryHostname()) {
```

Comment:
Why is this here?  Put this in ImageStreamImport validation, pass the allowed registries to the strategy, not the importer.